code/game/objects/items/devices/megaphone.dm

```diff
@@ -49,33 +67,42 @@
 	user.handle_speech_problems(message_pieces)
 	message = multilingual_to_message(message_pieces)
 	if((loc == user && !user.incapacitated()))
-		if(emagged)
-			if(insults)
-				saymsg(user, pick(insultmsg))
-				insults--
-			else
-				to_chat(user, "<span class='warning'>*BZZZZzzzzzt*</span>")
-		else
-			if(span)
-				message = "<span class='[span]'>[message]</span>"
-			saymsg(user, message)
+		if(HAS_TRAIT(src, TRAIT_CMAGGED))
+			message = pick(insultmsg)
+		saymsg(user, message)
 
-		spamcheck = TRUE
-		spawn(20)
-			spamcheck = FALSE
+		on_cooldown = TRUE
+		addtimer(VARSET_CALLBACK(src, on_cooldown, FALSE), (HAS_TRAIT(src, TRAIT_CMAGGED) || emagged) ? modified_cooldown : normal_cooldown)
 
-/obj/item/megaphone/proc/saymsg(mob/living/user as mob, message)
-	audible_message("<span class='game say'><span class='name'>[user.GetVoice()]</span> [user.GetAltName()] broadcasts, <span class='reallybig'>\"[message]\"</span></span>", hearing_distance = 14)
+/obj/item/megaphone/proc/saymsg(mob/living/user, message)
+	if(HAS_TRAIT(src, TRAIT_CMAGGED))
+		playsound(src, "sound/items/bikehorn.ogg", 50, TRUE)
+	else
+		playsound(src, "sound/items/megaphone.ogg", 100, FALSE, 5)
+
+	audible_message("<span class='game say'><span class='name'>[user.GetVoice()]</span> [user.GetAltName()] broadcasts, <span class='[span]'>\"[message]\"</span></span>", hearing_distance = 14)
 	log_say(message, user)
 	for(var/obj/O in oview(14, get_turf(src)))
-		O.hear_talk(user, message_to_multilingual("<span class='reallybig'>[message]</span>"))
+		O.hear_talk(user, message_to_multilingual("<span class='[span]'>[message]</span>"))
 
 	for(var/mob/M in get_mobs_in_view(7, src))
 		if((M.client?.prefs.toggles2 & PREFTOGGLE_2_RUNECHAT) && M.can_hear())
 			M.create_chat_message(user, message, FALSE, "big")
 
-/obj/item/megaphone/emag_act(user as mob)
-	if(!emagged)
-		to_chat(user, "<span class='warning'>You overload \the [src]'s voice synthesizer.</span>")
-		emagged = TRUE
-		insults = rand(1, 3)//to prevent dickflooding
+/obj/item/megaphone/cmag_act(mob/user)
+	if(HAS_TRAIT(src, TRAIT_CMAGGED))
+		return
+	if(user)
+		to_chat(user, "<span class='warning'>You drip some yellow ooze into [src]'s voice synthesizer, gunking it up.</span>")
+	playsound(src, "sparks", 75, TRUE, SHORT_RANGE_SOUND_EXTRARANGE)
+	ADD_TRAIT(src, TRAIT_CMAGGED, CLOWN_EMAG)
+
+/obj/item/megaphone/emag_act(mob/user)
+	if(emagged)
+		return
+	if(HAS_TRAIT(src, TRAIT_CMAGGED))  // one at a time
+		to_chat(user, "<span class='warning'>You don't want to gunk up your emag!</span>")
```

Comment:
to_chat here should indicate an action being done, "You short out [src]'s dampener circuits but it's covered in a weird ooze! You don't want to gunk up your emag!"
it would be better feedback